test/units/util.sh

```diff
@@ -110,3 +110,48 @@ create_dummy_container() {
     cp -a /testsuite-13-container-template/* "$root"
     coverage_create_nspawn_dropin "$root"
 }
+
+# Wait until file appears
+wait_for_file() {
+    local timeout=60
+    local file
+
+    if [[ $1 == '-t' ]]; then
+        timeout="$2"
+        shift 2
+    fi
+
+    file="$1"
+
+    timeout "$timeout" bash -c "while [[ ! -e $file ]]; do sleep .5; done"
+}
+
+# Wait until command succeeds
+wait_until() {
+    local timeout=60
+    local cmd
+
+    if [[ $1 == '-t' ]]; then
+        timeout="$2"
+        shift 2
+    fi
+
+    cmd="$*"
```

Comment:
It would be probably better to take the command as a string instead of an array, to avoid unnecessary evaluation along the way, i.e.

```suggestion
    cmd="${1:?}"
```

and then just call `wait_... -t 10 'foo bar baz'`.